pkg/controllers/dnspolicy/dnspolicy_healthchecks.go

```diff
@@ -32,21 +33,51 @@ func (r *DNSPolicyReconciler) reconcileHealthCheckProbes(ctx context.Context, dn
 		}
 	}
 
+	gwsWithIgnoredHosts := []string{}
 	// Reconcile DNSHealthCheckProbes for each gateway directly referred by the policy (existing and new)
 	for _, gw := range append(gwDiffObj.GatewaysWithValidPolicyRef, gwDiffObj.GatewaysMissingPolicyRef...) {
 		log.V(3).Info("reconciling probes", "gateway", gw.Name)
-		expectedProbes := r.expectedHealthCheckProbesForGateway(ctx, gw, dnsPolicy)
+		expectedProbes, ignoredHosts := r.expectedHealthCheckProbesForGateway(ctx, gw, dnsPolicy)
 		if err := r.createOrUpdateHealthCheckProbes(ctx, expectedProbes); err != nil {
 			return fmt.Errorf("error creating or updating expected probes for gateway %v: %w", gw.Gateway.Name, err)
 		}
 		if err := r.deleteUnexpectedGatewayHealthCheckProbes(ctx, expectedProbes, gw.Gateway, dnsPolicy); err != nil {
 			return fmt.Errorf("error removing unexpected probes for gateway %v: %w", gw.Gateway.Name, err)
 		}
-
+		if ignoredHosts != nil {
+			gwsWithIgnoredHosts = append(gwsWithIgnoredHosts, gw.Namespace+"/"+gw.Name)
+		}
 	}
+
+	addHealthCheckCondition(dnsPolicy, gwsWithIgnoredHosts)
+
 	return nil
 }
 
+func addHealthCheckCondition(dnsPolicy *v1alpha1.DNSPolicy, gwsWithIgnoredHosts []string) {
+	if dnsPolicy.Status.HealthCheck == nil {
+		dnsPolicy.Status.HealthCheck = &v1alpha1.HealthCheckStatus{}
+	}
+
+	noIgnoredHostsCondition := metav1.Condition{
+		Type:               DNSPolicyHealthChecksNoIgnoredHosts,
+		Status:             "True",
+		ObservedGeneration: dnsPolicy.Generation,
+		LastTransitionTime: metav1.Now(),
+		Reason:             DNSPolicyHealthChecksIgnoredReason,
+		Message:            "all gateways routes are creating health checks",
+	}
+
+	if len(gwsWithIgnoredHosts) > 0 {
+		noIgnoredHostsCondition.Status = "False"
+		noIgnoredHostsCondition.Reason = DNSPolicyHealthChecksIgnoredReason
+		noIgnoredHostsCondition.Message = "gateways with ignored wildcard hosts: " + strings.Join(gwsWithIgnoredHosts, ", ")
```

Comment:
This might be a bit noisy. Perhaps be better to just say, there are wildcard hosts/liseteners that have been ignored. Then leave it as an action of the user to look at the gateway definiton?